Add "Group Stats" for QM Object Cache Plugin

emrikol opened the pull request:
<!--
## For Automatticians!

:wave: Just a quick reminder that this is a public repo. Please don't include any internal links or sensitive data (like PII, private code, customer names, site URLs, etc. Any fixes related to security should be discussed with Platform before opening a PR. If you're not sure if something is safe to share, please just ask!

### BEFORE YOU PROCEED!!

If you’re editing a feature without changing the spirit of the implementation, fixing bugs, or performing upgrades, then please proceed!

If you’re adding a feature or changing the spirit of an existing implementation, please create a proposal in Cantina P2 using the MU Plugins Proposal Block Pattern. Please mention the [CODEOWNERS](.github/CODEOWNERS) of this repository and relevant stakeholders in your proposal :). Please be aware that any unplanned work may take some time to get reviewed. Thank you 🙇‍♀️🙇!

## For external contributors!

Welcome! We look forward to your contribution! ❤️
-->
## Description
<!--
A few sentences describing the overall goals of the Pull Request.

Should include any special considerations, decisions, and links to relevant GitHub issues.

Please don't include internal or private links :)
-->

These changes introduce a new feature to the Query Monitor plugin, which allows users to view detailed group statistics for the object cache operations. This can help in identifying the performance of different cache groups, and can provide useful insights for performance optimization.  Three files have been added or modified as follows:

* `class-qm-collector-object-cache-group-stats.php`: The `QM_Collector_Object_Cache_Group_Stats` collector class gathers statistics for the group data of object cache operations. It collects data such as operation count, time and size for each group.
* `class-qm-output-html-object-cache-group-stats.php`: The `QM_Output_Html_Object_Cache_Group_Stats` class is responsible for formatting and outputting the data collected by `QM_Collector_Object_Cache_Group_Stats`. It provides a human-readable output for the stats in the footer. It also includes helper methods to convert bytes into human-readable format, convert time to milliseconds, output sortable table columns, and handle table cell outputs.
* `qm-object-cache.php`: This file has been updated to include support for the new "Group Stats" collector and HTML output.

<img width="2274" alt="image" src="https://github.com/Automattic/vip-go-mu-plugins/assets/3060583/2977ddb3-5254-436b-aed3-e70d2a941ce1">

## Changelog Description

### Plugin Updated: qm-plugins

Added a "Group Stats" sub-panel for Object Cache to aid in development and debugging.

<!--
A description of the context of the change for a changelog. It should have a title, examples (if applicable), and why the change was made.

**Please keep the changelog title format same as in example below (### <Title>), as this is later used to generate the changelog entry title.**

Example for a plugin upgrade:

### Plugin Updated: Jetpack 9.2.1

We upgraded Jetpack 9.2 to Jetpack 9.2.1.

Not a lot of significant changes in this patch release, just bugfixes and compatibility improvements.
-->
## Pre-review checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change uses a rollout method to ease with deployment (if applicable - especially for large scale actions that require writes).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Pre-deploy checklist

- [ ] VIP staff: Ensure any alerts added/updated conform to internal standards (see internal documentation). 

## Steps to Test
<!--
Outline the steps to test and verify the PR here.

Example:

1. Check out PR.
1. Go to `wp-admin` > `Tools` > `Bakery`
1. Click on "Bake Cookies" button.
1. Verify cookies are delicious.
-->

emrikol committed:
Adding "Group Stats" Query Monitor panel

Adds a panel to view details about group stats to Query Monitor. Shows, per operation type: group item count, size, and time spent.

GaryJones commented on qm-plugins/qm-object-cache/html/class-qm-output-html-object-cache-group-stats.php:
```diff
@@ -0,0 +1,235 @@
+
+				echo '</tbody>';
+				echo '<tfoot>';
+				echo '<tr>';
```
I think there should be a corresponding `</tr>` somewhere below here.

emrikol replied:
Should be fixed now 🤞

emrikol committed:
Adding missing `</tr>`

rinatkhaziev reviewed (approved):
Thank you, this is great! 🎉

rinatkhaziev commented:
I did a round of basic testing, seems to work great

emrikol committed:
Properly validate array keys before adding

The current flawed method of validating array keys is flawed and throws mulitiple PHP warnings.

This should properly validate the array keys exist, and default them to `0` if they do not before trying to add to them.

emrikol committed:
Removing unused function.

`maybe_output_toggle_table_cell_for_array()` isn't used in this sub-panel and was left over from copying code from other panels 😬

WPprodigy committed:
Merge branch 'develop' into emrikol/qm-object-cache-group-stats
